zinit.zsh

```diff
@@ -1174,6 +1174,10 @@ builtin setopt noaliases
 # FUNCTION: @zinit-register-hook. [[[
 # Registers the z-annex inside Zinit – i.e. an Zinit extension
 @zinit-register-hook() {
+    emulate -LR zsh
+    # nobanghist to disable adding backslash escaping to "!", which is used in a lot of hocks and behaves differently
```

Comment:
```suggestion
    # Set 'nobanghist' option disables backslash escaping for "!"
```

How does it behave differently?

Reply:
Example:

```zsh
λ  cat excl-test
test=x\!y ; echo "${(q)test}"  

λ  source excl-test
x\!y

λ  zsh -c 'source excl-test'
x!y
```

(from a [comment in the bug report](https://github.com/zdharma-continuum/zinit/issues/199#issuecomment-1092191551))

Reply:
I hope to get the "how it behaves is differently" in the next line

```
# ...behaves differently 
# in interactive sessions (escapes !) and non-interactive ones (does not)
```

Reply:
Reworded it a bit for hopefully more readability